### PR TITLE
changes to laravel 10

### PR DESCRIPTION
**Pull request by juhniorsantos**

### CODE REVIEW

File: migrations/2017_09_17_000000_create_roles_table.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
 use Illuminate\Database\Schema\Blueprint;
 use Illuminate\Support\Facades\Schema;
 
-class CreateRolesTable extends Migration
+return new class extends Migration
```
Comment (rodrigopedra) [inline migrations/2017_09_17_000000_create_roles_table.php:7]: We can't make this change and keep `composer.json` allowing Laravel versions before 8, as support for anonymous migrations was introduced in Laravel 8

see: https://github.com/laravel/framework/pull/36906

File: migrations/2017_09_17_000000_create_roles_table.php [hunk 2]
```diff
@@ -19,6 +19,8 @@ public function up()
 
                 $table->unsignedSmallInteger('sort_index')->nullable();
 
+                $table->boolean('is_admin')->default(false);
```
Comment (rodrigopedra) [inline migrations/2017_09_17_000000_create_roles_table.php]: Where in the code is this new column used?

If adding this is required in a project level, one can publish the migrations and customize it for their needs

File: migrations/2017_09_17_100000_create_role_user_table.php [hunk 3]
```diff
@@ -23,7 +23,7 @@ public function up()
                 $userForeignKey = $userModel->getForeignKey();
 
                 $table->unsignedInteger('role_id');
-                $table->unsignedInteger($userForeignKey);
+                $table->unsignedBigInteger($userForeignKey);
```
Comment (rodrigopedra) [inline migrations/2017_09_17_100000_create_role_user_table.php:26]: We could consider moving all ids and foreign ids to use `unsignedBigInteger`, or use the more modern schema helpers

**Review by rodrigopedra — CHANGES_REQUESTED** [review]
Maybe it is better to work on a new major version to incorporate some of the suggestions.

Would requiring PHP 8.1 be an issue for you?

I can take a look at this later today. I am thinking on already making it Laravel 11 ready.

The only change I am not willing to incorporate is adding the `is_admin` column, unless it is needed in the package's code

File: migrations/2017_09_17_000000_create_roles_table.php (hunk 1 above)
Comment (rodrigopedra) [reply migrations/2017_09_17_000000_create_roles_table.php:7]: Removing previous Laravel versions would require a new major version

**Comment by juhniorsantos** [comment]
> Maybe it is better to work on a new major version to incorporate some of the suggestions.
> 
> Would requiring PHP 8.1 be an issue for you?
> 
> I can take a look at this later today. I am thinking on already making it Laravel 11 ready.
> 
> The only change I am not willing to incorporate is adding the `is_admin` column, unless it is needed in the package's code

PHP version 8.1 is not a problem, even 8.2 which will be a prerequisite for Laravel 11.

**Review by rodrigopedra — COMMENTED** [review]
Released version 1.0 with Laravel 9, 10 and 11 support, and PHP 8.1 requirement

**Comment by rodrigopedra** [comment]
Please note that the namespace of the `HasACL` was changed to `RodrigoPedra\LaravelSimpleACL\Concerns\HasACL`

Also, the default model shipped with the configuration was changed to `App\Models\User`, inline with modern Laravel releases

Thanks for the heads-up @juhniorsantos, let me known if you find any issues

**Pull request closed without merging**